### PR TITLE
Feature/6886 part1

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/6886

*Description of changes:*

- Create migration script of StaffTask table
- Implement get endpoint for the same
- Write unit tests
- WIP PR - next PR would contain inserting records in StaffTask during Org Pending process

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/tests/unit/api/test_task.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from auth_api import status as http_status
+from tests.utilities.factory_utils import factory_auth_header
+from tests.utilities.factory_scenarios import TestJwtClaims
+
+
+def test_fetch_tasks(client, jwt, session):  # pylint:disable=unused-argument
+    """Assert that the tasks can be fetched."""
+    headers = factory_auth_header(jwt=jwt, claims=TestJwtClaims.staff_role)
+    rv = client.get('/api/v1/tasks', headers=headers, content_type='application/json')
+    assert rv.status_code == http_status.HTTP_200_OK
```
Comment (saravanpa-aot) [inline auth-api/tests/unit/api/test_task.py:28]: assert the schema as well plz

File: auth-api/src/auth_api/services/task.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        task_schema = TaskSchema()
+        obj = task_schema.dump(self._model, many=False)
+        return obj
+
+    @staticmethod
+    def create_task(task_model: TaskModel):
+        """Create a new task record."""
+        current_app.logger.debug('<create_staff_task ')
+
+        task_model.add_to_session()
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/services/task.py:62]: i think save will add it to session as well

File: auth-api/src/auth_api/models/task.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class Task(BaseModel):
+    """Model for a Task record."""
+
+    __tablename__ = 'tasks'
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String(250), nullable=False)
+    date_submitted = Column(DateTime)
+    relationship_type = Column(String(50), nullable=False)
+    relationship_id = Column(Integer, index=True, nullable=False)
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/models/task.py:30]: I hope u really meant to put index on relationship_id

**Review by saravanpa-aot — APPROVED** [review]

File: auth-api/src/auth_api/resources/task.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    @staticmethod
+    @TRACER.trace()
+    @cors.crossdomain(origin='*')
+    @_JWT.has_one_of_roles(
+        [Role.STAFF.value])
+    def get():
+        """Fetch tasks."""
+        try:
+            response, status = json.dumps(StaffTaskService.fetch_tasks()), http_status.HTTP_200_OK
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/resources/task.py:45]: i think use ur marshamallow schema to serialise

File: auth-api/migrations/versions/c429d54cbe53_added_tasks_table.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    sa.Column('due_date', sa.DateTime(), nullable=True),
+    sa.Column('task_type', sa.String(length=50), nullable=False),
+    sa.Column('task_status', sa.String(length=50), nullable=False),
+    sa.Column('created_by_id', sa.Integer(), nullable=True),
+    sa.Column('modified_by_id', sa.Integer(), nullable=True),
+    sa.ForeignKeyConstraint(['created_by_id'], ['users.id'], ),
+    sa.ForeignKeyConstraint(['modified_by_id'], ['users.id'], ),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_index(op.f('ix_tasks_relationship_id'), 'tasks', ['relationship_id'], unique=False)
```
[the remark this answers is not in the document]
Comment (karthik-aot) [reply auth-api/migrations/versions/c429d54cbe53_added_tasks_table.py:38]: I was planning that we may have a case of filtering tasks by relationship id . But I would add id as an index as well

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**